source/jormungandr/jormungandr/street_network/here.py

```diff
@@ -598,102 +500,82 @@ def _create_matrix_response_with_direct_path(self, json_response, row):
 
-        # TODO handle max_duration (but the API can only handle a max distance (searchRange)
+        if post_resp.status_code != 202:
+            raise TechnicalError('Here, post return status code != 202')
 
-        return params
+        return post_resp.json()
 
-    def _get_sn_routing_matrix_with_simple_matrix(
-        self, origins, destinations, mode, max_duration, request, realtime_traffic, max_matrix_points
-    ):
-        params = self.get_matrix_params(
-            origins, destinations, mode, max_duration, request, realtime_traffic, max_matrix_points
-        )
-        r = self._call_here(self.matrix_service_url, params=params)
-        r.raise_for_status()
-        return self._create_matrix_response(r.json(), origins, destinations)
+    def get_matrix_response(self, origins, destinations, post_resp):
 
-    def _get_sn_routing_matrix_with_multi_direct_path(
-        self, origins, destinations, mode, max_duration, request, realtime_traffic, max_matrix_points
-    ):
-        sn_routing_matrix = response_pb2.StreetNetworkRoutingMatrix()
-        row = sn_routing_matrix.rows.add()
-        language = self.get_language_parameter(request)
-        for origin in itertools.islice(origins, int(max_matrix_points)):
-            for destination in itertools.islice(destinations, int(max_matrix_points)):
-                params = self.get_direct_path_params(
-                    origin,
-                    destination,
-                    mode,
-                    PeriodExtremity(datetime=request['datetime'], represents_start=request['clockwise']),
-                    request,
-                    realtime_traffic,
-                    language,
-                    self.engine_type,
-                    self.engine_average_consumption,
-                )
-                r = self._call_here(self.routing_service_url, params=params)
-                r.raise_for_status()
-                self._create_matrix_response_with_direct_path(r.json(), row)
-        return sn_routing_matrix
+        headers = {'Content-Type': 'application/json'}
+        matrix_id = post_resp.get('matrixId', None)
+        if matrix_id == None:
+            raise TechnicalError('Here, invalid matrixId inside matrix POST response')
+
+        # Here just expose a get to retreive matrix but you have to wait for the result to be available.
+        # It is your own waiting loop !
+        t = 0
+        step_time = self.lapse_time_matrix_to_retry
+        while t < self.timeout:
+            self.log.debug('Here, try to fetch matrix data with a get')
+            time.sleep(self.lapse_time_matrix_to_retry)
+            get_url = self.matrix_service_url + '/' + str(matrix_id) + '?apiKey=' + str(self.apiKey)
+            matrix_resp = requests.request("GET", get_url, headers=headers)
+            if matrix_resp.status_code == 200:
+                return self._create_matrix_response(matrix_resp.json(), origins, destinations)
+            t = t + step_time
```

Comment:
you can't use [Retrying](https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/tests/rabbitmq_utils.py#L105) instead of while

Reply:
ho thanks, It seems to be perfect !